### DISCUSSION TITLE
Using prisma with planetscale on cloudflare workers

### DISCUSSION THREAD

**Discussion by agcty** [prisma/prisma#16310] [Q&A] [answered]
I'm starting to dig into how the client and query engines work together and was thinking of a way to not have to use the prisma data proxy but connect to planetscale directly over their HTTP API (https://planetscale.com/blog/introducing-the-planetscale-serverless-driver-for-javascript).

Essentially what I'm thinking is, can't you write a custom query engine that takes the request from the client, translates it into SQL and simply uses the fetch API to make the query?

Planetscale already handles pooling etc. i don't necessarily need the data proxy I'm thinking, not because it's bad or anything but this seems like it could be a more direct/efficient way to access a planetscale db.

According to the prisma docs (https://www.prisma.io/docs/concepts/components/prisma-engines/query-engine) it seems this could work?

<img width="1038" alt="image" src="https://user-images.githubusercontent.com/12101091/202289721-a5215d1d-36f0-4e06-9246-1172852ebe56.png">

**Comment by janpio** [comment, marked as answer]
Feature request here: https://github.com/prisma/prisma/issues/15265
Unfortunately for us because of our architecture a multi month project, so will not happen soon.

**Reply by agcty** [reply]
awesome thanks! good to know it's something that's been thought of

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
